src/CppUTestExt/IEEE754ExceptionsPlugin.cpp

```diff
@@ -34,6 +34,53 @@ extern "C" {
     #include <fenv.h>
 }
 
+#if CPPUTEST_FENV_LACKS_SIGNALLING
+
+/*
+ * Missing implementations for Apple, MinGW and Watcom C.
+ */
+
+#if defined(__APPLE__)
+#define CONTROL_WORD __control
+#elif defined(__MINGW32__)
+#define CONTROL_WORD __control_word
+#elif defined(__WATCOMC__)
+#define CONTROL_WORD control_word
+#endif
+
+int feenableexcept (int excepts)
+{
+    fenv_t fenv;
+    unsigned short old_excepts;
+    unsigned short new_excepts = (unsigned short)excepts & FE_ALL_EXCEPT;
+
+    if (fegetenv (&fenv)) return -1;
+
+    old_excepts = fenv.CONTROL_WORD & FE_ALL_EXCEPT;
+    /* Mingw GCC promotes ~<unsigned short> to <int> and then complains about
+     * loss of information when casting from <int> to <unsigned short>. Yikes.
+     */
+    fenv.CONTROL_WORD &= (unsigned short)~new_excepts;
+
+    return (fesetenv(&fenv) ? -1 : old_excepts);
+}
+
+int fedisableexcept (int excepts)
+{
+    fenv_t fenv;
+    unsigned short old_excepts;
+    unsigned short new_excepts = (unsigned short)excepts & FE_ALL_EXCEPT;
```

Comment:
But this is really a conversion from int to unsigned short, so the Werror will be justified